pkg/pipeline/conntrack/hash_test.go

```diff
@@ -0,0 +1,212 @@
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipB, portB, ipA, portA, protocolA, 222, 11),
+			false,
+		},
+		{
+			"Alternating ip",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipB, portA, ipA, portB, protocolA, 222, 11),
+			false,
+		},
+		{
+			"Alternating port",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipA, portB, ipB, portA, protocolA, 222, 11),
+			false,
+		},
+		{
+			"Same IP+port, different protocol",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipA, portA, ipB, portB, protocolB, 222, 11),
+			false,
+		},
+	}
+	for _, test := range table {
+		t.Run(test.name, func(t *testing.T) {
+			h1, err1 := ComputeHash(test.flowLog1, keyFields)
+			h2, err2 := ComputeHash(test.flowLog2, keyFields)
+			require.NoError(t, err1)
+			require.NoError(t, err2)
+			if test.sameHash {
+				require.Equal(t, h1, h2)
+			} else {
+				require.NotEqual(t, h1, h2)
+			}
+		})
+	}
+}
+
+func TestComputeHash_Bidirectional(t *testing.T) {
+	keyFields := api.KeyFields{
+		FieldGroups: []api.FieldGroup{
+			{
+				Name: "src",
+				Fields: []string{
+					"SrcAddr",
+					"SrcPort",
+				},
+			},
+			{
+				Name: "dst",
+				Fields: []string{
+					"DstAddr",
+					"DstPort",
+				},
+			},
+			{
+				Name: "protocol",
+				Fields: []string{
+					"Proto",
+				},
+			},
+		},
+		Hash: api.ConnTrackHash{
+			FieldGroupRefs: []string{"protocol"},
+			FieldGroupARef: "src",
+			FieldGroupBRef: "dst",
+		},
+	}
+	ipA := "10.0.0.1"
+	ipB := "10.0.0.2"
+	portA := 1
+	portB := 9002
+	protocolA := 6
+	protocolB := 7
+	table := []struct {
+		name     string
+		flowLog1 config.GenericMap
+		flowLog2 config.GenericMap
+		sameHash bool
+	}{
+		{
+			"Same IP, port and protocol",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 222, 11),
+			true,
+		},
+		{
+			"Alternating ip+port",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipB, portB, ipA, portA, protocolA, 222, 11),
+			true,
+		},
+		{
+			"Alternating ip",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipB, portA, ipA, portB, protocolA, 222, 11),
+			false,
+		},
+		{
+			"Alternating port",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipA, portB, ipB, portA, protocolA, 222, 11),
+			false,
+		},
+		{
+			"Same IP+port, different protocol",
+			NewFlowLog(ipA, portA, ipB, portB, protocolA, 111, 22),
+			NewFlowLog(ipA, portA, ipB, portB, protocolB, 222, 11),
+			false,
+		},
+	}
+	for _, test := range table {
+		t.Run(test.name, func(t *testing.T) {
+			h1, err1 := ComputeHash(test.flowLog1, keyFields)
+			h2, err2 := ComputeHash(test.flowLog2, keyFields)
+			require.NoError(t, err1)
+			require.NoError(t, err2)
+			if test.sameHash {
+				require.Equal(t, h1, h2)
+			} else {
+				require.NotEqual(t, h1, h2)
+			}
+		})
+	}
+}
```

Comment:
What are we going to do with interfaces? if we want to see the flow only once, and we do not care from which interface we got it ???

Reply:
That's still an open question. In my current mental model, if we get multiple flow logs for the same flow from different interfaces, we can either add the interface field into the hash and get multiple connections or do not add it and aggregate the flow logs in the connection (which might result in wrong aggregation values)

Reply:
@ronensc so let's merge this as-is and think on that later. There are many options, and we will need to start with something that works. Let's plan to add the interface to the bi-directional hash. So this will get us separate flows for each of the interfaces. I think that for now, this is good. We can always decide to hook to only a sub-set of the interface if needed.

From my PoV there is not need to change this PR for this point ... if needed we will do that later.